src/test/regress/expected/multi_prepare_sql.out

```diff
@@ -1195,6 +1195,31 @@ EXECUTE countsome; -- no replanning
 ---------------------------------------------------------------------
 (0 rows)
 
+-- Standard planner converted text and varchar casts to cstring in some cases
+-- We make sure we convert it back to text when parsing the expression
+-- https://github.com/citusdata/citus/issues/6061
+-- https://github.com/citusdata/citus/issues/5646
+-- https://github.com/citusdata/citus/issues/5033
+CREATE TABLE test(t timestamptz, user_id int);
+SELECT create_distributed_table('test', 'user_id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+PREPARE test_statement(text) AS
+SELECT user_id FROM test WHERE t >= $1::timestamptz ORDER BY user_id;
+INSERT INTO test VALUES ('2022-02-02', 0);
+INSERT INTO test VALUES ('2022-01-01', 1);
+INSERT INTO test VALUES ('2021-01-01', 2);
+EXECUTE test_statement('2022-01-01');
```

Comment:
Executing prepared statements >=6 times is always useful (or even required) as Postgres does certain changes on the query after that